include/elements/3d/hexahedron_bspline_element.tcc

```diff
@@ -13,6 +14,20 @@ void mpm::HexahedronBSplineElement<Tdim, Tpolynomial>::
   //! Uniform spacing length in 3D
   this->spacing_length_ =
       std::abs(nodal_coordinates(1, 0) - nodal_coordinates(0, 0));
+
+  //! Identify if element is at boundary to see if kernel correction is
+  //! necessary
+  if (kernel_correction) {
+    for (unsigned n = 0; n < this->nconnectivity_; ++n) {
+      for (unsigned i = 0; i < Tdim; ++i) {
+        if (this->node_type_[n][i] == 1 || this->node_type_[n][i] == 4) {
+          this->kernel_correction_ = true;
+          goto breakout;
+        }
+      }
+    }
+  breakout:;
```

Comment:
The kernel correction needs to be applied in the full cell, right?

Reply:
Nope, only the elements in the vicinity of 1.5 * spacing_length_ away from the actual boundary.

Reply:
Okey, makes sense to me then